doc/userguide/performance/debug.rst

```diff
@@ -0,0 +1,67 @@
+Performance Debugging
+=====================
+
+There are many possibilities that could be the reason for performance issues.
+In this section we will guide you through some options.
+
+General
+-------
+
+First of all you should check all the log files with a focus on stats.log and
+suricata.log if any obvious issues are seen. There are several tools that can
+help to find a root cause.
+
+A first step is to run a tool like **htop** to get an overview of the system
+load and if there is a bottleneck with the traffic distribution. For example if
+you can see that only a small number of cpu cores hit 100% all the time and
+others don't, it could be related to a bad traffic distribution or elephant
+flows. In the first case try to improve the configuration, in the other case
+try to filter or shunt those big flows with either bpf filter, bypass rules or
+eBPF/XDP.
+
+Another helpful tool is **perf** which helps to spot performance issues. Make
+sure you have it installed and also the debug symbols installed for suricata or
+the output won't be very helpful. This output is also helpful when you report
+performance issues as the Suricata Development team can narrow down possible
+bugs with that.
+
+::
+
+    sudo perf top -p $(pidof suricata)
+
+If you see specific function calls at the top and red it's a hint that those
+are the bottlenecks. For example if you see **IPOnlyMatchPacket** it can be
+either a result of high drop rates or incomplete flows which result in
+decreased performance.
+
+Another recommendation is to run Suricata without any rules to see if it's
+mainly related to the traffic. It can also be helpful to use rule-profiling
+and/or packet-profiling at this step.
```

Comment:
It can be worth mentioning that for that part Suricata needs to be compiled with `enable-profiling` and that has a perf impact so it is advised not to leave it like that in prod.

Reply:
will add this note